csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/Razor.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Linq;
+
+namespace Semmle.Extraction.CSharp.DependencyFetching
+{
+    internal class Razor
+    {
+        private readonly DotnetVersion sdk;
+        private readonly ProgressMonitor progressMonitor;
+        private readonly DotNet dotNet;
+        private readonly string sourceGeneratorFolder;
+        private readonly string cscPath;
+
+        public Razor(DotnetVersion sdk, DotNet dotNet, ProgressMonitor progressMonitor)
+        {
+            this.sdk = sdk;
+            this.progressMonitor = progressMonitor;
+            this.dotNet = dotNet;
+
+            sourceGeneratorFolder = Path.Combine(this.sdk.FullPath, "Sdks", "Microsoft.NET.Sdk.Razor", "source-generators");
+            if (!Directory.Exists(sourceGeneratorFolder))
+            {
+                this.progressMonitor.RazorSourceGeneratorMissing(sourceGeneratorFolder);
+                throw new Exception($"Razor source generator folder {sourceGeneratorFolder} does not exist.");
+            }
+
+            cscPath = Path.Combine(this.sdk.FullPath, "Roslyn", "bincore", "csc.dll");
+            if (!File.Exists(cscPath))
+            {
+                this.progressMonitor.CscMissing(cscPath);
+                throw new Exception($"csc.dll {cscPath} does not exist.");
+            }
+        }
+
+        private static void GenerateAnalyzerConfig(IEnumerable<string> cshtmls, string analyzerConfigPath)
+        {
+            using var sw = new StreamWriter(analyzerConfigPath);
+            sw.WriteLine("is_global = true");
+
+            foreach (var f in cshtmls.Select(f => f.Replace('\\', '/')))
+            {
+                sw.WriteLine($"\n[{f}]");
+                var base64 = Convert.ToBase64String(Encoding.UTF8.GetBytes(f)); // TODO: this should be the relative path of the file.
```

Comment:
Why does it need to be the relative path?

Reply:
That's what the dotnet CLI is doing. I think this change would have some runtime consequences, but I'm unsure what. Also, it's not something that we currently rely on in any of the checks.